### ISSUE TITLE
calico 3.1.0 error calling getv: key does not exist

### ISSUE THREAD

**Issue by den-is** [projectcalico/calico#1906]
My setup running on virtualbox:
calico 3.1.0
k8s 1.10.1
etcd 3.2.18
centos7
default calico configs and manifests from docs
3 master nodes and 1 worker
kubm1.dev, kubm2.dev, kubm3.dev
kubw1.dev

Today when I started cluster for usual workflow everything worked but then in some amount of time, I've noticed that all my apps which require internal k8s networking are in CrashLoopBack (some php app, k8s-dashboard, coredns)
CoreDns can't reach api servers clusters.
I looked into calico logs and discovered never ending error log.

Restarting cluster helps for a few minutes then errors in calico are starting again.
Something triggers calico config update.

here is log output
https://pastebin.com/uLqDTUp8

For last few days I have not touched anything related to k8s networking or infrastructure.
I assume that something got corrupted because of often cluster shutdowns.
Absolutely same cluster setup is running as "production" sandbox environment without any issues

**Comment by tmjd** [comment]
Have you verified your etcd cluster is healthy and not corrupted and all 3 are still clustered together?

I see `2018-04-17 09:35:13.415 [WARNING][62] felix.go 933: Config change requires restart key="IpInIpTunnelAddr" old="10.22.199.64" updateType="delete"` in the pastebin that think the etcd endpoints might not be clustered together.  It may look like that key was removed if it is written to one etcd and then it later reads from a different one (and they aren't clustered or are degraded).

**Comment by den-is** [comment]
I'm sorry,
Because of urgency I had to destroy and rebuild this cluster. I haven't looked into etcd itself for health status.
Only thing I did, except monitoring logs for some time, was looking at `kubectl get componentstatus` which was showing everything as healthy.
Only assumption I have was - corrupted etcd.
Rebuilding exactly same setup doesn't has this issue.

Posted this issue just in case if you guys could notice something more important.

Closing this issue.
